src/blockchain.rs

```diff
@@ -23,9 +23,57 @@ pub struct Block {
     /// it must also contain its Receipt. Receipts appear in the order of their Transactions.
     pub receipts : Vec<Receipt>,
 }
+pub type BlockV1 = Block;
 
-impl Serializable for Block {}
-impl Deserializable for Block {}
+/// A data structure that describes and authorizes the execution of a batch of transactions (state transitions) on the blockchain.
+/// 
+/// [V1](Block) -> V2: 
+/// - "Header" is now of type BlockHeaderV2.
+/// - "Transactions" is now of type TransactionV2.
+/// - "Receipts" is now of type ReceiptV2.
+#[derive(BorshSerialize, BorshDeserialize, Clone)]
+pub struct BlockV2 {
+    /// Block header
+    pub header : BlockHeaderV2,
+
+    /// A dynamically sized list of Transactions.
+    pub transactions : Vec<TransactionV2>,
+
+    /// A dynamically sized list of Receipts. If a Block contains a
+    /// Transaction, it must also contain its Receipt. Receipts appear in the order of their
+    /// Transactions.
+    pub receipts : Vec<ReceiptV2>,
+}
+
+impl BlockV2 {
+    /// Conversion from [hotstuff_rs::types::Block], with an option to validate
+    /// signature of the transactions.
+    pub fn from_hotstuff_block(
+        block: hotstuff_rs::types::Block, validate_transactions: bool
+    ) -> Result<BlockV2, data::BlockConversionError> {
+        let blockdata = data::BlockDataV2::from_data(&block.data, validate_transactions)?;
+        Ok(BlockV2{
+            header: BlockHeaderV2 {
+                height: block.height,
+                hash: block.hash,
+                justify: block.justify,
+                data_hash: block.data_hash,
+                chain_id: blockdata.header.chain_id,
+                proposer: blockdata.header.proposer,
+                timestamp: blockdata.header.timestamp,
+                base_fee: blockdata.header.base_fee_per_gas,
```

Comment:
This field should be called “base_fee_per_gas” (and the field in BlockV1 as well).